onnxruntime/core/providers/shared/node_unit/node_unit.h

```diff
@@ -64,15 +64,16 @@ class NodeUnit {
   ProviderType GetExecutionProviderType() const noexcept;
 
   const Node& GetNode() const noexcept { return node_; }
-
+  const std::vector<const Node*> GetOutputNodes() const noexcept { return output_nodes_; }
   const std::vector<const Node*> GetAllNodes() const noexcept { return nodes_; }
 
  private:
-  std::vector<IODef> input_defs_;
-  std::vector<IODef> output_defs_;
+  std::vector<NodeUnitIODef> inputs_;
+  std::vector<NodeUnitIODef> outputs_;
 
-  const std::vector<const Node*> nodes_;  // all nodes in this NodeUnit
-  const Node& node_;                      // target Node
+  const std::vector<const Node*> nodes_;         // all nodes in this NodeUnit
+  const std::vector<const Node*> output_nodes_;  // all the nodes producing outputs for this NodeUnit
+  const Node& node_;                             // target Node
```

Comment:
>node_ [](http://example.com/codeflow?start=14&length=5)

how about the names `all_nodes_`, `nodes_producing_outputs_`, `target_node_`? then perhaps these comments aren't needed

Reply:
Updated the `node_` -> `target_node_`, and `nodes_` -> `all_nodes_`
Keep the `output_nodes_` and its comment to be consistent with the function `GetOutputNodes()`

Reply:
Seems we don't need `all_nodes_`, removed it for now, will add back if QDQ NodeUnit needs it (unlikely)